### PR TITLE
adding cpu and memory metrics

### PR DESCRIPTION
**Pull request by thugrock7**
## Description
Adding CPU and Memory monitoring metrics

### Testing
deployed and tested locally. 

### Checklist:
- [x] My changes generate no new warnings

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: instrumentation/opentelemetry/init.go
Comment (varkey98) [inline instrumentation/opentelemetry/init.go:438]: I think `InitializeSystemMetrics` would be better as we are calling it from here

**Comment by varkey98** [comment]
LGTM, @tim-mwangi pls take a look

**Review by varkey98 — APPROVED** [review]
Pls wait for a review from @tim-mwangi as well

File: instrumentation/opentelemetry/internal/metrics/system_metrics.go
Comment (puneet-traceable) [inline instrumentation/opentelemetry/internal/metrics/system_metrics.go:44]: please remove fmt based prints? I hope there should be a logger available in goagent.

Comment (tim-mwangi) [reply instrumentation/opentelemetry/internal/metrics/system_metrics.go:44]: We just use golangs `log`. There is no zap available. @thugrock7 eg. `log.Printf("error while shutting down tracer provider: %v\n", err)`

File: instrumentation/opentelemetry/internal/metrics/system_metrics.go
Comment (tim-mwangi) [inline instrumentation/opentelemetry/internal/metrics/system_metrics.go:96]: Let us prefix these metric names with `hypertrace.agent`

**Comment by tim-mwangi** [comment]
Strange that we do not get the reason for static analysis failure.

**Comment by tim-mwangi** [comment]
Let's make this configurable since in some deployments such as full TPA and ebpf, we already capture this. i.e can be disabled/enabled. We can add a `system_metrics_enabled` flag here https://github.com/hypertrace/agent-config/blob/1259231da9069021d09bcb6f3393275be0372582/proto/hypertrace/agent/config/v1/config.proto#L191

**Comment by varkey98** [comment]
@ryanericson @thugrock7 merging this as we discussed yesterday during standup call

**Pull request merged**
